nextcord/types/polls.py

```diff
@@ -0,0 +1,45 @@
+from typing import List, Literal, TypedDict
+
+from typing_extensions import NotRequired
+
+from .emoji import PartialEmoji
+
+LayoutType = Literal[1]
+
+
+class PollMedia(TypedDict):
+    text: str
+    emoji: NotRequired[PartialEmoji]
+
+
+class PollAnswer(TypedDict):
+    answer_id: NotRequired[int]
+    poll_media: PollMedia
+
+
+class PollCreateRequest(TypedDict):
+    question: PollMedia
+    answers: List[PollAnswer]
+    duration: int
+    allow_multiselect: bool
+    layout_type: NotRequired[LayoutType]
+
+
+class PollAnswerCount(TypedDict):
+    id: int
+    count: int
+    me_voted: bool
+
+
+class PollResults(TypedDict):
+    is_finalized: bool
+    answer_counts: List[PollAnswerCount]
+
+
+class Poll(TypedDict):
+    question: PollMedia
+    answers: List[PollAnswer]
```

Comment:
Hmm, wouldn't it be better to use a dataclass with the ``to_dict`` method for http stuff instead of TypedDict?
It would provide a way to retrieve attributes through `.` + Additionally, you can add `__repr__`.